main.py

```diff
@@ -948,19 +954,49 @@ def notify_switch_enabled(self, dpid):
         event = KytosEvent(name=name, content={'dpid': dpid})
         self.controller.buffers.app.put(event)
 
+    def notify_switch_links_enable(self, switch):
+        """Send an event to notify the status of a link in a switch"""
+        for interface in switch.interfaces.values():
```

Comment:
@Alopalao it should work as you've implemented, if we cover the case when iterating over this mutable shared collection from a threaded method, however if you iterate over interfaces you'll end up with duplicated Links, since we also need to notify only one (I mentioned more details about this in another comment, check it out), so I believe if you were to iterate over `self.links` with the `self._links_lock` then that'd simplify a bit, iterating once and without sets to make it unique, since you could check if either interface belongs to the switch.

Reply:
Changed.